### PR TITLE
Add in filterrific for filtering.

### PR DESCRIPTION
**Pull request by seanmarcia**
Improved filtering for children index page.

Added in the `filterrific` gem that we can now use elsewhere in the app. 

This also updates the `rubocop-rails` gem and fixes the deprecations it had. See giphy below:

Filter

### CODE REVIEW

File: app/views/children/index.html.erb [hunk 1]
```diff
@@ -30,72 +30,62 @@
...
-                    <%= collection_select(:filters, :from_family, @family || {}, :id, :guardian_display_name, {include_blank: true, selected: @selected_family}, class: "form-control") %>
-                  </div>
-                <% end %>
-                <% if @children.present? %>
-                  <div class="form-group col-lg-4 col-md-4 col-sm-6 col-xs-12">
-                    <%= label_tag "by Children First Name", "Child's First Name" %>
-                    <%= collection_select(:filters, :from_children, @children || {}, :id, :first_name, {include_blank: true, selected: @selected_children_first_name}, class: "form-control") %>
-                  </div>
-                <% end %>
+                <div class="col-2">
```
Comment (edwinthinks) [inline app/views/children/index.html.erb:35]:
```suggestion
                <div class="col-3">
```
I noticed when I tested it locally the labels were a bit squished (maybe due to my laptop?)
With `col-2`
Screen Shot 2020-06-19 at 10 10 21 AM

With `col-3`
Screen Shot 2020-06-19 at 10 10 35 AM

File: app/views/children/index.html.erb [hunk 2]
```diff
@@ -30,72 +30,62 @@
...
+                </div>
+                <div class="col-2">
+                  <%= f.label :search_families, "Search By Guardian Name" %>
+                  <%= f.text_field(
+                          :search_families,
+                          class: 'filterrific-periodically-observed form-control'
+                      ) %>
+                </div>
+                <div class="col-2">
+                  <%= f.label :search_comments, "Search By Active Children" %>
```
Comment (edwinthinks) [inline app/views/children/index.html.erb:50]:
```suggestion
                  <%= f.label :search_comments, "Show Active Only?" %>
```
A minor suggestion here. I read "Search By Active Children" to sort of implying that the Child is active. I think making 'Show Active Only?" might be a bit clearer though I can see the original working too :)

File: Gemfile.lock [hunk 3]
```diff
@@ -445,6 +459,7 @@ DEPENDENCIES
   guard-rspec
   jbuilder (~> 2.5)
   jquery-rails (~> 4.3, >= 4.3.1)
+  kaminari
```
Comment (edwinthinks) [inline Gemfile.lock:462]: Are we currently using  `kaminari` in this PR?

File: app/views/children/index.html.erb (hunk 1 above)
Comment (seanmarcia) [reply app/views/children/index.html.erb:35]: I'll make the change! I'm on a 38 monitor and I imagine some of the diaper banks are not on giant monitors too!

File: app/views/children/index.html.erb (hunk 2 above)
Comment (seanmarcia) [reply app/views/children/index.html.erb:50]: Changed!

File: Gemfile.lock (hunk 3 above)
Comment (seanmarcia) [reply Gemfile.lock:462]: Kaminari is needed for filterrific :)

**Pull request merged**